src/components/datetime.rs

```diff
@@ -18,9 +18,38 @@ use tinystr::TinyAsciiStr;
 use super::{
     calendar::{CalendarDateLike, GetTemporalCalendar},
     duration::normalized::{NormalizedTimeDuration, RelativeRoundResult},
-    Date, Duration, Time,
+    Date, Duration, MonthCode, PartialDate, Time,
 };
 
+/// A partial DateTime record
+#[derive(Debug, Default, Copy, Clone)]
+pub struct PartialDateTime {
+    // A potentially set `year` field.
+    pub year: Option<i32>,
+    // A potentially set `month` field.
+    pub month: Option<i32>,
+    // A potentially set `month_code` field.
+    pub month_code: Option<MonthCode>,
+    // A potentially set `day` field.
+    pub day: Option<i32>,
+    // A potentially set `era` field.
+    pub era: Option<TinyAsciiStr<16>>,
+    // A potentially set `era_year` field.
+    pub era_year: Option<i32>,
+    // A potentially set `hour` field.
+    pub hour: Option<i32>,
+    // A potentially set `minute` field.
+    pub minute: Option<i32>,
+    // A potentially set `second` field.
+    pub second: Option<i32>,
+    // A potentially set `millisecond` field.
+    pub millisecond: Option<i32>,
+    // A potentially set `microsecond` field.
+    pub microsecond: Option<i32>,
+    // A potentially set `nanosecond` field.
+    pub nanosecond: Option<i32>,
```

Comment:
Thinking that this should reuse `PartialDate` and `PartialTime` instead.

Reply:
Oh, I think I misread the above a bit. I definitely prefer having the internal be `PartialDate` and `PartialTime`, so I'll flip it back